example/middleware/index.dart

```diff
@@ -1,34 +1,35 @@
-import 'dart:html';
 import 'dart:async';
+import 'dart:html';
 
 import 'package:redux/redux.dart';
 
-render(int state) {
-  querySelector('#value').innerHtml = '${state}';
+void render(int state) {
+  querySelector('#value').innerHtml = '$state';
 }
 
+enum Actions { increment, decrement }
+
 // Create a Reducer with a State (int) and an Action (String) Any dart object
 // can be used for Action and State.
-int counterReducer(int state, action) {
-  switch (action) {
-    case 'INCREMENT':
-      return state + 1;
-    case 'DECREMENT':
-      return state - 1;
-    default:
-      return state;
+int counterReducer(int state, dynamic action) {
```

Comment:
To reinforce good practices, primitive state should not be shown. It is an option, yes, but state should not be mutated in the reducer.

So while this is possible, I suggest reinforcing best practices.

Reply:
Just to make sure I understand -- how are we mutating the `state` that's passed in? We're returning the result of the expression `state + 1` or `- 1`. Would that mutate the state object passed in?

Reply:
From react:
>  * The shape of the state is up to you: it can be a primitive, an array, an object,
>  * or even an Immutable.js data structure. The only important part is that you should
>  * not mutate the state object, but return a new object if the state changes.

IMHO, using a primitive leads people to bad state objects where the reducer manipulates the value vs mutating it.

From this example, I'd feel file saying `mymodel.value +1` as opposed to creating a new object with the new value. I say this because I literally did that with Distinctly and I knew better.

A comment could be enough for sure, though.

Reply:
Ah, good insight, thanks for sharing.

I'll keep the examples as-is and add comments. Since they are simple & mirror the example on the original Redux.js repo, I think they might be a familiar starting point for folks  coming from JS-land, but comments would help reinforce the point you make.